### PR TITLE
Fix: Share Button working

### LINKED ISSUE #72: feat: Implement share app feature in the application [GCI]
**Summary:**

Add the share app feature in the application

**Steps to reproduce:**

- Open the application
- Click on `Share` in the navigation drawer
- Nothing happens

**Expected behaviour:**

The task here is to add this functionality

### PR DESCRIPTION
**Pull request by rachittt**
Fixes #72 

GIF:
shareFeature

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: app/src/main/kotlin/org/mifos/mobile/cn/ui/mifos/DashboardActivity.kt [hunk 1]
```diff
@@ -185,6 +185,13 @@ class DashboardActivity : MifosBaseActivity(), View.OnClickListener, NavigationV
             R.id.item_settings -> {
                 replaceFragment(SettingsFragment.newInstance(), true, R.id.container)
             }
+            R.id.item_share -> {
+                val sharingIntent = Intent(android.content.Intent.ACTION_SEND)
+                sharingIntent.type = "text/plain"
+                sharingIntent.putExtra(android.content.Intent.EXTRA_SUBJECT, "App Link")
```
Comment (SaumyaSingh1) [inline app/src/main/kotlin/org/mifos/mobile/cn/ui/mifos/DashboardActivity.kt]: @rachittt  put the string "App Link" in strings.xml

File: app/src/main/kotlin/org/mifos/mobile/cn/ui/mifos/DashboardActivity.kt [hunk 2]
```diff
@@ -185,6 +185,13 @@ class DashboardActivity : MifosBaseActivity(), View.OnClickListener, NavigationV
...
+                sharingIntent.putExtra(android.content.Intent.EXTRA_TEXT,(R.string.share_message))
+                startActivity(Intent.createChooser(sharingIntent, "Share the app link via:"))
+            }
         }
```
Comment (SaumyaSingh1) [inline app/src/main/kotlin/org/mifos/mobile/cn/ui/mifos/DashboardActivity.kt]: @rachittt  remove blank lines

File: app/src/main/kotlin/org/mifos/mobile/cn/ui/mifos/DashboardActivity.kt [hunk 3]
```diff
@@ -185,6 +185,13 @@ class DashboardActivity : MifosBaseActivity(), View.OnClickListener, NavigationV
             R.id.item_settings -> {
                 replaceFragment(SettingsFragment.newInstance(), true, R.id.container)
             }
+            R.id.item_share -> {
+                val sharingIntent = Intent(android.content.Intent.ACTION_SEND)
+                sharingIntent.type = "text/plain"
+                sharingIntent.putExtra(android.content.Intent.EXTRA_SUBJECT, "App Link")
+                sharingIntent.putExtra(android.content.Intent.EXTRA_TEXT,(R.string.share_message))
+                startActivity(Intent.createChooser(sharingIntent, "Share the app link via:"))
```
Comment (SaumyaSingh1) [inline app/src/main/kotlin/org/mifos/mobile/cn/ui/mifos/DashboardActivity.kt]: @rachittt  use strings.xml for strings

**Comment by SaumyaSingh1** [comment]
@rachittt  Make these small changes : )

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -310,6 +310,8 @@
     <string name="privacy_policy_host" translatable="false">openmf.github.io/privacy_policy_mifos_mobile.html</string>
     <string name="privacy_policy_host_url" translatable="false">https://openmf.github.io/privacy_policy_mifos_mobile.html</string>
     <string name="opensource_license_title">Open Source licenses</string>
-
+        <string name="app_link">App Link</string>
```
Comment (luckyman20) [inline app/src/main/res/values/strings.xml:313]: spacing problem

Comment (rachittt) [reply app/src/main/res/values/strings.xml]: thanks sir!

**Pull request merged**